README.rst

```diff
@@ -4,24 +4,35 @@
 .. image:: https://img.shields.io/pypi/pyversions/twine.svg
     :target: https://pypi.org/project/twine
 
-.. image:: https://img.shields.io/readthedocs/twine
-    :target: https://twine.readthedocs.io
-
 .. image:: https://img.shields.io/github/workflow/status/pypa/twine/Main
     :target: https://github.com/pypa/twine/actions
 
 .. image:: https://img.shields.io/codecov/c/github/pypa/twine
     :target: https://codecov.io/gh/pypa/twine
 
+.. docs-badges-end
+
+.. image:: https://img.shields.io/readthedocs/twine
+    :target: https://twine.readthedocs.io
+
 twine
 =====
 
+.. docs-intro-start
+
 Twine is a utility for `publishing`_ Python packages on `PyPI`_.
 
 It provides build system independent uploads of source and binary
-`distribution artifacts <distributions_>`_ for both new and existing
+`distribution artifacts`_ for both new and existing
 `projects`_.
 
+.. _`publishing`: https://packaging.python.org/tutorials/packaging-projects/
+.. _`PyPI`: https://pypi.org
+.. _`distribution artifacts`: https://packaging.python.org/glossary/#term-Distribution-Package
+.. _`projects`: https://packaging.python.org/glossary/#term-Project
```

Comment:
I find this separation of links confusing both in `README.rst` (because they're not with the rest of the ones at the bottom) and in `index.rst` (because at least `PyPI` is used more than once, and the include means it's not obvious where it's defined).

However, this did point out that the `term` links are broken: https://github.com/pypa/twine/issues/788.

Reply:
I saw some places where blocks of links are defined after each block/paragraph. Of course, it's not ideal if they are reused multiple times but it's also useful that they are close to their use context.